File: android2_lesson01/src/java/java/com/example/ama/android2_lesson01/db/base/NotesDataManager.java
package com.example.ama.android2_lesson01.db.base;

import com.example.ama.android2_lesson01.model.Note;

import java.util.ArrayList;

/**
 * Interface describes methods for managing notes
 */

public interface NotesDataManager {
    void loadListOfAllNotes(LoadDataCallback callback);

    void createNote(String title, String text, DataChangedCallback callback);

    void updateNote(Note targetNote, String newTitle, String newText, DataChangedCallback callback);

    void deleteNote(Note targetNote, DataChangedCallback callback);

    interface LoadDataCallback {
        void onLoad(ArrayList<Note> mData);
    }

    interface DataChangedCallback {
        void onDataChanged();
    }
}

File: android2_lesson01/src/kotlin/java/com/example/ama/android2_lesson01/db/base/NotesDataManager.kt
package com.example.ama.android2_lesson01.db.base

import com.example.ama.android2_lesson01.model.Note

/**
 * Interface describes methods for managing notes
 */

interface NotesDataManager {
    fun loadListOfAllNotes(callback: LoadDataCallback)
    fun createNote(title: String, text: String, callback: DataChangedCallback)
    fun updateNote(targetNote: Note, newTitle: String, newText: String, callback: DataChangedCallback)
    fun deleteNote(targetNote: Note, callback: DataChangedCallback)

    interface LoadDataCallback {
        fun onLoad(mData: ArrayList<Note>)
    }

    interface DataChangedCallback {
        fun onDataChanged()
    }
}